apis/v1alpha1/shared_types.go

```diff
@@ -133,15 +133,15 @@ type AdminNetworkPolicyIngressPeer struct {
 	// Support: Core
 	//
 	// +optional
-	Namespaces *NamespacedPeer `json:"namespaces,omitempty"`
+	Namespaces *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
```

Comment:
Field name (`Namespaces`) and json field name (`namespaceSelector`) don't match.

With this PR, we have:
- `Namespaces` in `AdminNetworkPolicySubject`
- `NamespaceSelector` in `NamespacedPod`
- (not clear which one you meant) in `AdminNetworkPolicyIngressPeer`/`AdminNetworkPolicyEgressPeer`

This seems like it could be confusing... it was definitely less confusing with `NamespacedPeer` there because then a peer had either

    - namespaces:
        namespaceSelector: ...

or

    - pods:
        podSelector: ...

Maybe we should keep `NamespacedPeer` even though it's mostly vestigial, just to preserve the symmetry? (Or maybe not? Maybe it's annoying of us to force users to have to include that extra level?)

(If we do that here, we should probably do the same thing in Subject).

Reply:
yeah so I wanted to make this the same as what's in the subject today.. so don't make users go through that extra step/hoop for no reason, 
btw nice catch on the naming diff ..
Let me rebase this PR

Reply:
al'right so in the latest update this is what I did:

subject and peer are symmetric now in how we express namespaces and pods:
```
namespaces:
   matchLabels:
```
```
pods:
   namespaces:
     matchLabels:
   pods:
     matchLabels:
```